File: parker-senter/who-likes-it.js
function likes(names) {
    //if no likes
    let thumbsUp = names.length;
    if (thumbsUp === 0) {
      return 'no one likes this';
    }
    //if one like
    else if (thumbsUp === 1) {
      return names[0] + ' likes this';
    }
    //if two likes
    else if (thumbsUp === 2) {
      return names[0] + ' and ' + names[1] + ' like this';
    }
    //if three likes
    else if (thumbsUp === 3) {
      return names[0] + ', ' + names[1] + ' and ' + names[2] + ' like this'
    }
    //if four or more likes
    else if (thumbsUp >= 4) {
      return names[0] + ', ' + names[1] + ' and ' + (thumbsUp -2) + ' others like this'
    }
  }
  
  /*
  PSEUDOCODE
  //check for names in array
  //tell who has liked the post
  
  FUNC:
  var thumbsUp = names.length
  if (thumbsUp === # of likes) {
    return string that correlates
  }
  
  */

File: parker-senter/who_likes_it.py
def likes(names):
    #checks if theres no likes
    if len(names) == 0:
        return 'no one likes this'
    #checks if theres one like
    elif len(names) == 1:
        return names[0] + ' likes this'
    #checks if theres 2 likes
    elif len(names) == 2:
        return names[0] + ' and ' + names[1] + ' like this'
    #checks if theres 3 likes
    elif len(names) == 3:
        return names[0] + ', ' + names[1] + ' and ' + names[2] + ' like this'
    #checks if theres 4 or more likes
    elif len(names) >= 4:
        return names[0] + ', ' + names[1] + ' and ' + str(len(names) - 2) + ' others like this'
    pass

"""
PSEUDOCODE
take in names list
check the length of names
if there is 0 return no likes
if 1 name return that one person likes it
if 2 names return 2 people like it
if 3 names return 3 people like it 
if 4 or more return first 2 names and how ever many others like it
"""
